app/src/main/java/com/capstone/weatherapp/State.kt

```diff
@@ -0,0 +1,7 @@
+package com.capstone.weatherapp
+
+sealed class State<out T> {
```

Comment:
instead of templatizing this, let's create a sealed class for the Group state and one for the Details state. this way, we can fill in all the details and there won't be any conflicts between the two pages